neo/test/coretest/test_analogsignal.py

```diff
@@ -498,6 +498,79 @@ def test_as_quantity(self):
         self.assertIsInstance(sig_as_q, pq.Quantity)
         assert_array_equal(self.data1, sig_as_q.magnitude.flat)
 
+    def test_splice_1channel_inplace(self):
+        signal_for_splicing = AnalogSignal([0.1, 0.1, 0.1],
+                                           t_start=3*pq.ms,
+                                           sampling_rate=self.signal1.sampling_rate,
+                                           units=pq.uA)
+        result = self.signal1.splice(signal_for_splicing, copy=False)
+        assert_array_equal(result.magnitude.flatten(),
+                           np.array([0.0, 1.0, 2.0, 100.0, 100.0, 100.0, 6.0, 7.0, 8.0, 9.0]))
+        assert_array_equal(self.signal1, result)  # in-place
+        self.assertEqual(result.segment, self.signal1.segment)
```

Comment:
it would be nice to also check the channel_index here as well as in the 'with_copy' test below